src/perf_tool_backend.ml

```diff
@@ -412,35 +447,35 @@ module Recording = struct
       | Some (_, exit) -> perf_exit_to_or_error exit
       | _ -> Ok ()
     in
-    ( { pid = perf_pid; when_to_snapshot }
+    ( { pid = perf_pid; snapshot_behavior }
     , { Data.callgraph_mode = selected_callgraph_mode } )
   ;;
 
   let maybe_take_snapshot t ~source =
-    let signal =
-      match t.when_to_snapshot, source with
+    let should_take_snapshot =
+      match t.snapshot_behavior, source with
       (* [`never] only comes up in [-full-execution] mode. In that mode, perf always gives a
          complete trace; there's no snapshotting. *)
-      | `never, _ -> None
+      | Never, _ -> false
       (* Do not snapshot at the end of a program if the user has set up a trigger symbol. *)
-      | `function_call, `ctrl_c -> None
+      | Function_call _, `ctrl_c -> false
       (* This shouldn't happen unless there was a bug elsewhere. It would imply that a trigger
          symbol was hit when there is no trigger symbol configured. *)
-      | `at_exit _, `function_call -> None
+      | At_exit _, `function_call -> false
       (* Trigger symbol was hit, and we're configured to look for them. *)
-      | `function_call, `function_call -> Some Signal.usr2
+      | Function_call _, `function_call -> true
       (* Ctrl-C was hit, and we're configured to look for that. *)
-      | `at_exit signal, `ctrl_c ->
-        (* The actual signal to use varies depending on whether or not the user's version of perf
-           supports snapshot-at-exit. *)
-        Some
-          (match signal with
-           | `sigint -> Signal.int
-           | `sigusr2 -> Signal.usr2)
+      | At_exit _, `ctrl_c -> true
     in
-    match signal with
-    | None -> ()
-    | Some signal -> Signal_unix.send_i signal (`Pid t.pid)
+    if should_take_snapshot
+    then (
+      match t.snapshot_behavior with
+      | Never -> failwith "unreachable"
```

Comment:
Suggestion: lift this block into a `let snapshot () = ...`, then have the `-> (true|false)` branches above be `-> (() | snapshot ())`. That way we don't have to double-state the `Never` case.